plugins/catalog-backend/src/descriptors/model/envelope.ts

```diff
@@ -33,20 +27,12 @@ const descriptorEnvelopeSchema: yup.Schema<DescriptorEnvelope> = yup
   })
   .noUnknown();
 
+// Validate some raw structured data as a descriptor envelope
 export async function parseDescriptorEnvelope(
-  rawYaml: string,
+  data: any,
```

Comment:
Could bring in a `JsonData` type, with all the possible json values.

Reply:
Hm mmm worth it? Going by what the parser takes as input here. Restricting input to a validation function of this sort may be overkill?

Reply:
Yeah just thinking it's something we might want anyway